java-checks/src/main/java/org/sonar/java/checks/AssertionsInTestsCheck.java

```diff
@@ -93,18 +95,21 @@
   private final Deque<Boolean> methodContainsAssertion = new ArrayDeque<>();
   private final Deque<Boolean> inUnitTest = new ArrayDeque<>();
   private final Map<Symbol, Boolean> assertionInMethod = new HashMap<>();
+  private final Set<Symbol> visitedMethods = new HashSet<>();
   private JavaFileScannerContext context;
 
   @Override
   public void scanFile(final JavaFileScannerContext context) {
     this.context = context;
     assertionInMethod.clear();
+    visitedMethods.clear();
     inUnitTest.push(false);
     methodContainsAssertion.push(false);
     scan(context.getTree());
     methodContainsAssertion.pop();
     inUnitTest.pop();
     assertionInMethod.clear();
+    visitedMethods.clear();
```

Comment:
As discussed, you can probably rely on the keySet of assertionInMethod map rather than using an other set.